File: fixtures/rust_output/multipart_with_headers.rs
extern crate reqwest;
use reqwest::{header, multipart};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut headers = header::HeaderMap::new();
    headers.insert("Authorization", "Bearer ACCESS_TOKEN".parse().unwrap());
    headers.insert("X-Nice", "Header".parse().unwrap());

    let form = multipart::Form::new()
        .text("attributes", "{\"name\":\"tigers.jpeg\", \"parent\":{\"id\":\"11446498\"}}")
        .file("file", "myfile.jpg")?;

    let res = reqwest::Client::new()
        .post("https://upload.box.com/api/2.0/files/content")
        .headers(headers)
        .multipart(form)
        .send()?
        .text()?;
    println!("{}", res);

    Ok(())
}

File: test/fixtures/php-guzzle/multipart_with_headers.php
<?php
require 'vendor/autoload.php';

use GuzzleHttp\Client;
use GuzzleHttp\Psr7;

$client = new Client();

$response = $client->post('http://localhost:28139/api/2.0/files/content', [
    'headers' => [
        'Authorization' => 'Bearer ACCESS_TOKEN',
        'X-Nice'        => 'Header'
    ],
    'multipart' => [
        [
            'name' => 'attributes',
            'contents' => '{"name":"tigers.jpeg", "parent":{"id":"11446498"}}'
        ],
        [
            'name' => 'file',
            'contents' => Psr7\Utils::tryFopen('myfile.jpg', 'r')
        ]
    ]
]);
